### PR TITLE
[Source] Bump scripts to v0.1.0

### PR DESCRIPTION
**Pull request by aiyengar2**
Do not merge till https://github.com/rancher/rke2-charts/pull/60 is merged.

### CODE REVIEW

**Comment by aiyengar2** [comment]
TLDR: Migration should be good to go. No breaking changes for this repository

Integration test results:

```bash
arvindiyengar: ~/Rancher/charts-build-scripts/src/github.com/rancher/charts-build-scripts 
$ REPOSITORY=https://github.com/rancher/rke2-charts.git BRANCH=main-source ./scripts/integration
Pulling in https://github.com/rancher/rke2-charts.git at main-source
> Detecting type of branch
Branch main-source is a source branch
> Running make prepare
./bin/charts-build-scripts prepare
INFO[0000] Local chart does not need to be prepared     
INFO[0000] Loading dependencies for chart               
INFO[0000] Pulling https://charts.helm.sh/stable/packages/coredns-1.10.1.tgz from upstream into charts 
INFO[0000] Loading dependencies for chart               
INFO[0000] Applying changes from generated-changes      
INFO[0000] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrole-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrole.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrolebinding-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrolebinding.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/configmap-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/configmap.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/deployment-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/poddisruptionbudget.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/podsecuritypolicy.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/service-metrics.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/service.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/serviceaccount-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/serviceaccount.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/servicemonitor.yaml.patch 
INFO[0000] Applying: generated-changes/patch/values.yaml.patch 
INFO[0000] Pulling https://github.com/kubernetes/ingress-nginx/releases/download/ingress-nginx-3.3.0/ingress-nginx-3.3.0.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-createSecret.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-patchWebhook.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/controller-daemonset.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/controller-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/default-backend-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
INFO[0001] Local chart does not need to be prepared     
INFO[0001] Loading dependencies for chart               
INFO[0001] Pulling https://charts.helm.sh/stable/packages/metrics-server-2.11.1.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/metrics-server-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
> Running make patch
./bin/charts-build-scripts patch
INFO[0000] Local chart does not need to be patched      
INFO[0000] Pulling https://charts.helm.sh/stable/packages/coredns-1.10.1.tgz from upstream into charts-original 
INFO[0000] Loading dependencies for chart               
INFO[0000] Generating changes to generated-changes      
INFO[0000] Patch: generated-changes/patch/Chart.yaml    
INFO[0000] Patch: generated-changes/patch/templates/_helpers.tpl 
INFO[0000] Patch: generated-changes/patch/templates/clusterrole-autoscaler.yaml 
INFO[0000] Patch: generated-changes/patch/templates/clusterrole.yaml 
INFO[0000] Patch: generated-changes/patch/templates/clusterrolebinding-autoscaler.yaml 
INFO[0000] Patch: generated-changes/patch/templates/clusterrolebinding.yaml 
INFO[0000] Patch: generated-changes/patch/templates/configmap-autoscaler.yaml 
INFO[0000] Patch: generated-changes/patch/templates/configmap.yaml 
INFO[0000] Patch: generated-changes/patch/templates/deployment-autoscaler.yaml 
INFO[0000] Patch: generated-changes/patch/templates/deployment.yaml 
INFO[0000] Patch: generated-changes/patch/templates/poddisruptionbudget.yaml 
INFO[0000] Patch: generated-changes/patch/templates/podsecuritypolicy.yaml 
INFO[0000] Patch: generated-changes/patch/templates/service-metrics.yaml 
INFO[0000] Patch: generated-changes/patch/templates/service.yaml 
INFO[0000] Patch: generated-changes/patch/templates/serviceaccount-autoscaler.yaml 
INFO[0000] Patch: generated-changes/patch/templates/serviceaccount.yaml 
INFO[0000] Patch: generated-changes/patch/templates/servicemonitor.yaml 
INFO[0000] Patch: generated-changes/patch/values.yaml   
INFO[0000] Pulling https://github.com/kubernetes/ingress-nginx/releases/download/ingress-nginx-3.3.0/ingress-nginx-3.3.0.tgz from upstream into charts-original 
INFO[0000] Loading dependencies for chart               
INFO[0000] Generating changes to generated-changes      
INFO[0000] Patch: generated-changes/patch/Chart.yaml    
INFO[0000] Patch: generated-changes/patch/templates/_helpers.tpl 
INFO[0000] Patch: generated-changes/patch/templates/admission-webhooks/job-patch/job-createSecret.yaml 
INFO[0000] Patch: generated-changes/patch/templates/admission-webhooks/job-patch/job-patchWebhook.yaml 
INFO[0000] Patch: generated-changes/patch/templates/controller-daemonset.yaml 
INFO[0000] Patch: generated-changes/patch/templates/controller-deployment.yaml 
INFO[0000] Patch: generated-changes/patch/templates/default-backend-deployment.yaml 
INFO[0000] Patch: generated-changes/patch/values.yaml   
INFO[0000] Local chart does not need to be patched      
INFO[0000] Pulling https://charts.helm.sh/stable/packages/metrics-server-2.11.1.tgz from upstream into charts-original 
INFO[0001] Loading dependencies for chart               
INFO[0001] Generating changes to generated-changes      
INFO[0001] Patch: generated-changes/patch/Chart.yaml    
INFO[0001] Patch: generated-changes/patch/templates/_helpers.tpl 
INFO[0001] Patch: generated-changes/patch/templates/metrics-server-deployment.yaml 
INFO[0001] Patch: generated-changes/patch/values.yaml   
> Running make clean
./bin/charts-build-scripts clean
> Checking if Git is clean...
> Running make validate
./bin/charts-build-scripts validate
INFO[0000] Validating against released charts in main   
INFO[0000] Local chart does not need to be prepared     
INFO[0000] Loading dependencies for chart               
INFO[0000] Generated archive: assets/rke2-canal/rke2-canal-v3.13.300-build2021022302-rc00.tgz 
INFO[0000] Generated chart: charts/rke2-canal/rke2-canal/v3.13.300-build2021022302-rc00 
INFO[0000] Pulling https://charts.helm.sh/stable/packages/coredns-1.10.1.tgz from upstream into charts 
INFO[0000] Loading dependencies for chart               
INFO[0000] Applying changes from generated-changes      
INFO[0000] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrole-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrole.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrolebinding-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/clusterrolebinding.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/configmap-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/configmap.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/deployment-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/poddisruptionbudget.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/podsecuritypolicy.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/service-metrics.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/service.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/serviceaccount-autoscaler.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/serviceaccount.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/servicemonitor.yaml.patch 
INFO[0000] Applying: generated-changes/patch/values.yaml.patch 
INFO[0000] Generated archive: assets/rke2-coredns/rke2-coredns-1.10.101-build2021022302-rc00.tgz 
INFO[0000] Generated chart: charts/rke2-coredns/rke2-coredns/1.10.101-build2021022302-rc00 
INFO[0000] Pulling https://github.com/kubernetes/ingress-nginx/releases/download/ingress-nginx-3.3.0/ingress-nginx-3.3.0.tgz from upstream into charts 
INFO[0000] Loading dependencies for chart               
INFO[0000] Applying changes from generated-changes      
INFO[0000] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0000] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-createSecret.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-patchWebhook.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/controller-daemonset.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/controller-deployment.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/default-backend-deployment.yaml.patch 
INFO[0000] Applying: generated-changes/patch/values.yaml.patch 
INFO[0000] Generated archive: assets/rke2-ingress-nginx/rke2-ingress-nginx-3.3.001-rc00.tgz 
INFO[0000] Generated chart: charts/rke2-ingress-nginx/rke2-ingress-nginx/3.3.001-rc00 
INFO[0000] Local chart does not need to be prepared     
INFO[0000] Loading dependencies for chart               
INFO[0000] Generated archive: assets/rke2-kube-proxy/rke2-kube-proxy-v1.20.5-build2021031801-rc00.tgz 
INFO[0000] Generated chart: charts/rke2-kube-proxy/rke2-kube-proxy/v1.20.5-build2021031801-rc00 
INFO[0000] Pulling https://charts.helm.sh/stable/packages/metrics-server-2.11.1.tgz from upstream into charts 
INFO[0000] Loading dependencies for chart               
INFO[0000] Applying changes from generated-changes      
INFO[0000] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0000] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0000] Applying: generated-changes/patch/templates/metrics-server-deployment.yaml.patch 
INFO[0000] Applying: generated-changes/patch/values.yaml.patch 
INFO[0000] Generated archive: assets/rke2-metrics-server/rke2-metrics-server-2.11.100-build2021022301-rc00.tgz 
INFO[0000] Generated chart: charts/rke2-metrics-server/rke2-metrics-server/2.11.100-build2021022301-rc00 
INFO[0000] Pulling rancher/rke2-charts from upstream into original-assets 
INFO[0001] Found the following latest release candidate versions: {
 "rke2-canal/v3.13.300-build2021022302": "rke2-canal/v3.13.300-build2021022302-rc00",
 "rke2-coredns/1.10.101-build2021022302": "rke2-coredns/1.10.101-build2021022302-rc00",
 "rke2-ingress-nginx/3.3.001": "rke2-ingress-nginx/3.3.001-rc00",
 "rke2-kube-proxy/v1.20.5-build2021031801": "rke2-kube-proxy/v1.20.5-build2021031801-rc00",
 "rke2-metrics-server/2.11.100-build2021022301": "rke2-metrics-server/2.11.100-build2021022301-rc00"
} 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-canal/rke2-canal-v3.13.300-build2021022302.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-canal/rke2-canal/v3.13.300-build2021022302 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-coredns/rke2-coredns-1.10.101-build2021022302.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-coredns/rke2-coredns/1.10.101-build2021022302 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-ingress-nginx/rke2-ingress-nginx-3.3.001.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-ingress-nginx/rke2-ingress-nginx/3.3.001 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-kube-proxy/rke2-kube-proxy-v1.20.5-build2021031801.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-kube-proxy/rke2-kube-proxy/v1.20.5-build2021031801 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-metrics-server/rke2-metrics-server-2.11.100-build2021022301.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-metrics-server/rke2-metrics-server/2.11.100-build2021022301 
INFO[0001] Successfully validated against main!         
> Checking if Git is clean...
Successfully validated changes do not break source branch.
arvindiyengar: ~/Rancher/charts-build-scripts/src/github.com/rancher/charts-build-scripts 
$ REPOSITORY=https://github.com/rancher/rke2-charts.git BRANCH=main ./scripts/integration 
Pulling in https://github.com/rancher/rke2-charts.git at main
> Detecting type of branch
Branch main is a live branch
> Running sync on the following:
- url: https://github.com/rancher/rke2-charts.git
  branch: main-source
  dropReleaseCandidates: true
> Running sync on the following:
> Running make validate
./bin/charts-build-scripts validate
> Running make sync
./bin/charts-build-scripts sync
INFO[0000] Synchronizing with charts that will be generated from main-source 
INFO[0000] Pulling rancher/rke2-charts from upstream into new-assets 
INFO[0001] Local chart does not need to be prepared     
INFO[0001] Loading dependencies for chart               
INFO[0001] Generated archive: assets/rke2-canal/rke2-canal-v3.13.300-build2021022302-rc00.tgz 
INFO[0001] Generated chart: charts/rke2-canal/rke2-canal/v3.13.300-build2021022302-rc00 
INFO[0001] Pulling https://charts.helm.sh/stable/packages/coredns-1.10.1.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/clusterrole-autoscaler.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/clusterrole.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/clusterrolebinding-autoscaler.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/clusterrolebinding.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/configmap-autoscaler.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/configmap.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/deployment-autoscaler.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/poddisruptionbudget.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/podsecuritypolicy.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/service-metrics.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/service.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/serviceaccount-autoscaler.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/serviceaccount.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/servicemonitor.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
INFO[0001] Generated archive: assets/rke2-coredns/rke2-coredns-1.10.101-build2021022302-rc00.tgz 
INFO[0001] Generated chart: charts/rke2-coredns/rke2-coredns/1.10.101-build2021022302-rc00 
INFO[0001] Pulling https://github.com/kubernetes/ingress-nginx/releases/download/ingress-nginx-3.3.0/ingress-nginx-3.3.0.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-createSecret.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/admission-webhooks/job-patch/job-patchWebhook.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/controller-daemonset.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/controller-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/default-backend-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
INFO[0001] Generated archive: assets/rke2-ingress-nginx/rke2-ingress-nginx-3.3.001-rc00.tgz 
INFO[0001] Generated chart: charts/rke2-ingress-nginx/rke2-ingress-nginx/3.3.001-rc00 
INFO[0001] Local chart does not need to be prepared     
INFO[0001] Loading dependencies for chart               
INFO[0001] Generated archive: assets/rke2-kube-proxy/rke2-kube-proxy-v1.20.5-build2021031801-rc00.tgz 
INFO[0001] Generated chart: charts/rke2-kube-proxy/rke2-kube-proxy/v1.20.5-build2021031801-rc00 
INFO[0001] Pulling https://charts.helm.sh/stable/packages/metrics-server-2.11.1.tgz from upstream into charts 
INFO[0001] Loading dependencies for chart               
INFO[0001] Applying changes from generated-changes      
INFO[0001] Applying: generated-changes/patch/Chart.yaml.patch 
INFO[0001] Applying: generated-changes/patch/templates/_helpers.tpl.patch 
INFO[0001] Applying: generated-changes/patch/templates/metrics-server-deployment.yaml.patch 
INFO[0001] Applying: generated-changes/patch/values.yaml.patch 
INFO[0001] Generated archive: assets/rke2-metrics-server/rke2-metrics-server-2.11.100-build2021022301-rc00.tgz 
INFO[0001] Generated chart: charts/rke2-metrics-server/rke2-metrics-server/2.11.100-build2021022301-rc00 
INFO[0001] Found the following latest release candidate versions: {
 "rke2-canal/v3.13.300-build2021022302": "rke2-canal/v3.13.300-build2021022302-rc00",
 "rke2-coredns/1.10.101-build2021022302": "rke2-coredns/1.10.101-build2021022302-rc00",
 "rke2-ingress-nginx/3.3.001": "rke2-ingress-nginx/3.3.001-rc00",
 "rke2-kube-proxy/v1.20.5-build2021031801": "rke2-kube-proxy/v1.20.5-build2021031801-rc00",
 "rke2-metrics-server/2.11.100-build2021022301": "rke2-metrics-server/2.11.100-build2021022301-rc00"
} 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-canal/rke2-canal-v3.13.300-build2021022302.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-canal/rke2-canal/v3.13.300-build2021022302 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-coredns/rke2-coredns-1.10.101-build2021022302.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-coredns/rke2-coredns/1.10.101-build2021022302 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-ingress-nginx/rke2-ingress-nginx-3.3.001.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-ingress-nginx/rke2-ingress-nginx/3.3.001 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-kube-proxy/rke2-kube-proxy-v1.20.5-build2021031801.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-kube-proxy/rke2-kube-proxy/v1.20.5-build2021031801 
INFO[0001] Generated archive: new-assets/assets-without-rc/rke2-metrics-server/rke2-metrics-server-2.11.100-build2021022301.tgz 
INFO[0001] Generated chart: new-assets/charts-without-rc/rke2-metrics-server/rke2-metrics-server/2.11.100-build2021022301 
INFO[0002] Sync was successful!                         
INFO[0002] Successfully synchronized with main-source!  
INFO[0002] Creating or updating the Helm index with the newly added assets... 
INFO[0002] Nothing to sync. Working directory is up to date. 
Successfully validated changes do not break live branch.
```

**Comment by aiyengar2** [comment]
@dweomer we need to merge https://github.com/rancher/rke2-charts/pull/60 before merging this PR.

File: scripts/regenerate-packages [hunk 1]
```diff
@@ -6,10 +6,11 @@ cd $(dirname $0)
 if [[ -z ${BRANCH} ]]; then
     branch=$(git rev-parse --abbrev-ref HEAD)
 else
-    echo "Using branch ${BRANCH}"
     branch=${BRANCH}
 fi
 
+echo "Using branch ${branch}"
```
Comment (briandowns) [inline scripts/regenerate-packages:12]: Does this variable name need to be capitalized?

Comment (aiyengar2) [reply scripts/regenerate-packages:12]: we copy BRANCH into branch, so the variable we want to use is the lowercase version.

Also, this script is pretty much never going to be used by this repository again; it's just something that gets updated by the automated `make docs` command to keep it in sync with the Go templates in rancher/charts-build-scripts, but it won't affect the functionality of this repository at all.

**Review by briandowns — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**